Error response from daemon: client version 1.40 is too new

I am using latest Docker for Mac 2.0.0.3, Engine: 18.09.2.

```
=== RUN   TestNginxLatestReturn
--- FAIL: TestNginxLatestReturn (0.01s)
    app_test.go:23: failed to create container: creating reaper failed: Error response from daemon: client version 1.40 is too new. Maximum supported API version is 1.39
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1739281]

goroutine 36 [running]:
testing.tRunner.func1(0xc000141900)
	/usr/local/Cellar/go/1.12/libexec/src/testing/testing.go:830 +0x388
panic(0x17fc540, 0x1fdf940)
	/usr/local/Cellar/go/1.12/libexec/src/runtime/panic.go:522 +0x1b5
```

--- a/docker.go
+++ b/docker.go
@@ -187,6 +187,7 @@ func NewDockerProvider() (*DockerProvider, error) {
 	if err != nil {
 		return nil, err
 	}
+	client.NegotiateAPIVersion(context.Background())
 	p := &DockerProvider{
 		client: client,
 	}

--- a/go.mod
+++ b/go.mod
@@ -5,7 +5,7 @@ require (
 	github.com/Microsoft/go-winio v0.4.11 // indirect
 	github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5 // indirect
 	github.com/docker/distribution v2.7.0+incompatible // indirect
-	github.com/docker/docker v0.7.3-0.20180815000130-e05b657120a6
+	github.com/docker/docker v0.7.3-0.20190506211059-b20a14b54661
 	github.com/docker/go-connections v0.4.0
 	github.com/docker/go-units v0.3.3 // indirect
 	github.com/go-sql-driver/mysql v1.4.1

